docs/developers/blobstream-rollup-outline.md

```diff
@@ -0,0 +1,326 @@
+the Blobstream specific structure called the `Span`. The goal of these
+structures are to locate the data in the Celestia block so that we can prove
+that data's inclusion via Blockstream if needed. More information on namespaces
+can be found
+[here](https://github.com/celestiaorg/celestia-app/blob/v1.1.0/specs/src/specs/namespace.md),
+but think of this like a chain ID. More information about `shares` can be found
+[here](https://github.com/celestiaorg/celestia-app/blob/v1.1.0/specs/src/specs/shares.md),
+these are small chunks of the encoded Celestia block. We use the same encoding
+here so that the commitments to the rollup block match those committed to by
+validators in the Celestia data root.
+
+The `Span` could take many forms, but in this demo we will use the following:
+
+```go
+// Span describes the location of the rollup block data that is posted to
+// Celestia. This is important for other nodes to be able to prove that data in
+// the celestia block. This can be thought of as a pointer to some data in the Celestia block.
+type Span struct {
+    // CelestiaHeight is the height of the Celestia block that contains the
+    // rollup block data.
+    CelestiaHeight uint64 `json:"celestia_height"`
+    // DataShareStart is the index of the first share of the rollup block data.
+    DataShareStart uint64 `json:"share_start"`
+    // DataShareLen is lenght in shares of the rollup block data. This is used
+    // to identify all of the rollup block data in a Celestia block.
+    DataShareLen uint64 `json:"share_end"`
+}
+```
+
+We can then define the blockchain as a collection of blocks and some additional
+information about the chain such as the sequencer address.
+
+```go
+type Blockchain struct {
+    Blocks          []Block
+    SequncerAddress []byte
+    Namespace       []byte
+}
+```
+
+## Rollup Sequencer
+
+The rollup sequencer is responsible for creating blocks and, in this demo,
+writing that data to Celestia and Ethereum. The rollup fullnode is responsible
+for reading that data from Celestia and Ethereum and verifying that it follows
+the protocol rules of that rollup.
+
+Therefore we can start by first defining the reading and writing interations
+rollup nodes will have with both the Celestia and Ethereum networks. The actual
+implementations of these interfaces are left as excercises to the reader (🤪).
+Assume that those implementations of these interfaces are verifying the
+respective chain. For the connection to Celestia, this would likely mean
+connecting to a Celestia Light Node, which can detect faults in consensus such
+as hidden data. For the connection to Ethereum, this would likely mean running
+and connecting to a fullnode. More information on the RPC that is exposed by a
+Celestia Light Node can be found
+[here](https://node-rpc-docs.celestia.org/?version=v0.11.0), and more
+information on running one can be found
+[here](https://docs.celestia.org/nodes/light-node/).
```

Comment:
```suggestion
in the [running a light node documentation](https://docs.celestia.org/nodes/light-node/).
```

Reply:
same with this one 😄 https://github.com/celestiaorg/docs/blob/f8cb1a50eca74d663d3fd19a3679c7274fe7707e/docs/developers/blobstream-offchain.md?plain=1#L107